tests/golden_tests/compile_term/nums.hvm

```diff
@@ -0,0 +1 @@
+(+ 0xFFFF_FFFF 0b101)
```

Comment:
Can you also add a test for wrong number syntax? Like using decimal numbers when specifying binary and using hexa when specifying decimal